src/miral/basic_window_manager.cpp

```diff
@@ -153,6 +153,9 @@ auto miral::BasicWindowManager::add_surface(
         break;
     }
 
+    if (window_info.type() == mir_window_type_anchored)
+        place_anchored(window_info);
```

Comment:
1. I think this logic belongs in place_new_surface() i.e. that policy->place_new_window() has the final say.
2. place_anchored() checks the type() - so the "if ()" here is redundant.